Firmware/Marlin_main.cpp

```diff
@@ -2820,15 +2820,13 @@ static void gcode_G80()
     run = false;
 #endif //PINDA_THERMISTOR
 
-    uint8_t nMeasPoints = code_seen('N') ? code_value_uint8() : eeprom_read_byte((uint8_t*)EEPROM_MBL_POINTS_NR);
-    if (nMeasPoints != 7) {
-        nMeasPoints = 3;
-    }
-
-    uint8_t nProbeRetryCount = code_seen('C') ? code_value_uint8() : eeprom_read_byte((uint8_t*)EEPROM_MBL_PROBE_NR);
-    if (nProbeRetryCount > 10) {
-        nProbeRetryCount = 10;
-    }
+    uint8_t nMeasPoints = eeprom_read_byte((uint8_t*)EEPROM_MBL_POINTS_NR);
+    if (code_seen('N') && (code_value_uint8() == 7 || code_value_uint8() == 3))
+      nMeasPoints = code_value_uint8();
+    
+    uint8_t nProbeRetryCount = eeprom_read_byte((uint8_t*)EEPROM_MBL_PROBE_NR);
+    if (code_seen('C') && code_value_uint8() >= 1 && code_value_uint8() <= 10)
+      nProbeRetryCount = code_value_uint8();
```

Comment:
This code can be optimised more :) 

This below shaves off 14 bytes, so the PR only adds 6 bytes instead of 20

![image](https://github.com/prusa3d/Prusa-Firmware/assets/8218499/2fac7590-82da-4594-828b-755b483db510)

Reply:
This way `code_value_uint8()` is called less frequently and value scope is limited to the if statement

Reply:
code_seen needs to be called before code_value. It sets the pointer. So I had to change it up a bit. Still saves 12 bytes.